EMMItem.cs

```diff
@@ -21,11 +22,46 @@ public sealed class EMMItem : GlobalItem
 
 		public Modifier Modifier;
 
+		// Special item parameters
+		public bool hasRolled;             // whether this item has rolled a modifier TODO: Save this
```

Comment:
hasRolled is great and should stay.
However, effects that may influence an item do not belong in EMMItem.
Anything that influenced a rolled item should go in a separate GlobalItem to keep the code clean.
Feel free to make a new GlobalItem, it can be named ModifierItem.